tracker/tracker-assist/src/ScreenRecordingState.ts

```diff
@@ -0,0 +1,149 @@
+import ConfirmWindow from './ConfirmWindow/ConfirmWindow.js'
+import { recordRequestDefault, } from './ConfirmWindow/defaults.js'
+import type { Options as AssistOptions, } from './Assist'
+
+export enum RecordingState {
+  Off,
+  Requested,
+  Recording,
+}
+
+const defaultStyles = '2px dashed red'
+const leftTop = { left: 0, top: 0, position: 'fixed', }
+const bottomRight = { right: 0, bottom: 0, position: 'fixed', }
+
+const borderEmulationStyles = {
+  left: {
+    ...leftTop,
+    height: '100vh',
+    width: 0,
+    borderLeft: defaultStyles,
+  },
+  top: {
+    ...leftTop,
+    height: 0,
+    width: '100vw',
+    borderTop: defaultStyles,
+  },
+  right: {
+    ...bottomRight,
+    height: '100vh',
+    width: 0,
+    borderRight: defaultStyles,
+  },
+  bottom: {
+    ...bottomRight,
+    height: 0,
+    width: '100vw',
+    borderBottom: defaultStyles,
+  },
+}
+
+export default class ScreenRecordingState {
+  public status = RecordingState.Off
+  private agentsRecordingSession: string[] = []
+  private overlayAdded = false
+
+  constructor(
+    private readonly onAccept: () => void,
+    private readonly onDeny: () => void,
+    private readonly options: AssistOptions
+  ) {}
+
+  public get isActive() {
+    return this.status !== RecordingState.Off
+  }
+
+  private confirm: ConfirmWindow | null = null
+
+  public requestRecording = (id: string) => {
+    if (this.status !== RecordingState.Off) return
```

Comment:
this won't add `id` to the `agentsRecordingSession`  for the any agent other than first.

Btw, do we really need allow several agents to make recording simultaneously?

Reply:
The `agentsRecordingSession `  logic is unclear.
Right now it is something in between - `agentsRecordingSession` is array, but only one agen is able to start recording.
Changes required depending on the answer of the question above.

Reply:
unclear `agentsRecordingSession ` HERE